src/gamestate/commands.cpp

```diff
@@ -3844,6 +3844,62 @@ void execute_toggle_rebel_hunting(sys::state& state, dcon::nation_id source, dco
 	}
 }
 
+void toggle_select_province(sys::state& state, dcon::nation_id source, dcon::province_id prov) {
+	payload p;
+	memset(&p, 0, sizeof(payload));
+	p.type = command_type::toggle_select_province;
+	p.source = source;
+	p.data.generic_location.prov = prov;
+	add_to_command_queue(state, p);
+}
+bool can_toggle_select_province(sys::state& state, dcon::nation_id source, dcon::province_id prov) {
+	if(state.world.province_get_nation_from_province_control(prov) != source)
+		return false;
+	if(state.world.province_get_nation_from_province_ownership(prov) != source)
+		return false;
+	return true;
+}
+void execute_toggle_select_province(sys::state& state, dcon::nation_id source, dcon::province_id prov) {
+	if(!can_toggle_select_province(state, source, prov))
+		return;
+
+	for(auto m : state.world.province_get_current_modifiers(prov)) {
+		if(m.mod_id == state.economy_definitions.selector_modifier) {
+			sys::remove_modifier_from_province(state, prov, m.mod_id);
+			return;
+		}
+	}
+	sys::add_modifier_to_province(state, prov, state.economy_definitions.selector_modifier, sys::date{});
```

Comment:
similarly, add modifier to province is going to loop over them to ensure that the modifier is not already present